applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c

```diff
@@ -446,6 +442,26 @@ static struct bt_bap_broadcast_sink_cb broadcast_sink_cbs = {
 	.syncable = syncable_cb,
 };
 
+static void bt_mgmt_evt_handler(const struct zbus_channel *chan)
+{
+	const struct bt_mgmt_msg *msg;
+
+	msg = zbus_chan_const_msg(chan);
+
+	switch (msg->event) {
+	case BT_MGMT_BROADCAST_CODE_RECEIVED:
+		LOG_DBG("Broadcast code received");
+		LOG_HEXDUMP_DBG("Broadcast code", BT_ISO_BROADCAST_CODE_SIZE, msg->bcode);
+		memcpy(bis_encryption_key, msg->bcode, BT_ISO_BROADCAST_CODE_SIZE);
+		break;
+
+	default:
+		break;
+	}
+}
+
+ZBUS_LISTENER_DEFINE(bt_mgmt_evt_listener, bt_mgmt_evt_handler);
+
```

Comment:
I don't think there should be a zbus listener here. Main can rather set the broadcast_code with a `set()` function